packages/ui-navigation/src/Navigation/NavigationItem/index.tsx

```diff
@@ -75,15 +75,14 @@ class NavigationItem extends Component<NavigationItemProps> {
     const props = omitProps(this.props, NavigationItem.allowedProps)
 
     return (
-      //@ts-expect-error TODO: INSTUI-3245
       <ElementType
         {...props}
         href={href}
         onClick={onClick}
         css={this.props.styles?.navigationItem}
-        aria-label={this.props.minimized ? label : undefined}
+        aria-label={this.props.minimized ? (label as string) : undefined}
         // @ts-expect-error TODO: fix TS2590: Expression produces a union type that is too complex to represent.
-        ref={(element) => {
+        ref={(element: Element | null) => {
```

Comment:
Something is wrong here in the original code. If we allow `label` to be a React node then the screenreader will read garbage for the `aria-label`. We should only allow `string` for the label prop.

This would be a breaking change, but luckily no one uses this component currently.